Source/Core/Common/CompatPatches.cpp

```diff
@@ -0,0 +1,121 @@
+	void* low_heap = nullptr;
+	for (; !low_heap && target_addr <= highest_addr; target_addr += 0x1000) {
+		low_heap = VirtualAlloc((void*)target_addr, max_heap_size, MEM_RESERVE, PAGE_READWRITE);
+	}
+	if (!low_heap)
+		return nullptr;
+	return RtlCreateHeap(0, low_heap, 0, 0, nullptr, nullptr);
+}
+
+static bool ModifyProtectedRegion(void *address, size_t size, std::function<void()> func) {
+	DWORD old_protect;
+	if (!VirtualProtect(address, size, PAGE_READWRITE, &old_protect))
+		return false;
+	func();
+	if (!VirtualProtect(address, size, old_protect, &old_protect))
+		return false;
+	return true;
+}
+
+// Does not do input sanitization - assumes well-behaved input since Ldr has already parsed it.
+class ModuleParser
+{
+public:
+	ModuleParser(uintptr_t module_base) : base(module_base) {
+		auto mz = reinterpret_cast<PIMAGE_DOS_HEADER>(base);
+		auto pe = reinterpret_cast<PIMAGE_NT_HEADERS>(base + mz->e_lfanew);
+		directories = pe->OptionalHeader.DataDirectory;
+	}
+	template <typename T>
+	T GetRva(uint32_t rva) {
+		return reinterpret_cast<T>(base + rva);
+	}
+	bool PatchIAT(const char *module_name, const char *function_name, void* value) {
+		auto import_dir = &directories[IMAGE_DIRECTORY_ENTRY_IMPORT];
+		for (auto import_desc = GetRva<PIMAGE_IMPORT_DESCRIPTOR>(import_dir->VirtualAddress);
+			import_desc->OriginalFirstThunk; import_desc++) {
+			auto module = GetRva<const char *>(import_desc->Name);
+			auto names = GetRva<PIMAGE_THUNK_DATA>(import_desc->OriginalFirstThunk);
+			auto thunks = GetRva<PIMAGE_THUNK_DATA>(import_desc->FirstThunk);
+			if (!stricmp(module, module_name)) {
+				for (auto name = names; name->u1.Function; name++) {
+					if (!IMAGE_SNAP_BY_ORDINAL(name->u1.Ordinal)) {
+						auto import = GetRva<PIMAGE_IMPORT_BY_NAME>(name->u1.AddressOfData);
+						if (!strcmp(import->Name, function_name)) {
+							auto index = name - names;
+							return ModifyProtectedRegion(&thunks[index], sizeof(thunks[index]), [=] {
+								thunks[index].u1.Function = reinterpret_cast<decltype(thunks[index].u1.Function)>(value);
+							});
+						}
+					}
+				}
+				// Function not found
+				return false;
+			}
+		}
+		// Module not found
+		return false;
+	}
+private:
+	uintptr_t base;
+	PIMAGE_DATA_DIRECTORY directories;
+};
+
+void InstallCompatPatches(LdrWatcher *watcher) {
+  watcher->Install({ { L"EZFRD64.dll", L"811EZFRD64.DLL" }, [](const LdrDllLoadEvent& event) {
+    // *EZFRD64 is incldued in software packages for cheapo third-party gamepads (and gamepad adapters).
+    // The module cannot handle its heap being above 4GB, which tends to happen very often on modern Windows.
+    // NOTE: The patch will always be applied, but it will only actually avoid the crash if applied
+    // before module initialization (i.e. called on the Ldr callout path).
+    auto parser = ModuleParser(event.base_address);
+    parser.PatchIAT("kernel32.dll", "HeapCreate", HeapCreateLow4GB);
+  } });
+}
```

Comment:
It seems the linter hit an error: `fatal: Invalid symmetric difference expression eefae402b3ebf86a05861ec3cde057905dbfff87...` @leoetlino any idea? Not that it's a big deal, I'm just using this PR to generate test builds for users at the moment...with fix up style later if it works.